marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/reader/HubDataSourceReader.java

```diff
@@ -126,6 +145,20 @@ private int determineNumPartitions(Map<String, String> options) {
         return numPartitions;
     }
 
+    private CustomReadApiDefinitions readCustomJobApiDefinitions(Map<String, String> options) {
```

Comment:
Because there's only one API endpoint, there's no need for a CustomReadApiDefinition class. CustomWriteApiDefinitions was created as a holder for two API definitions. We don't have that need here, so this method should just return a single InputStreamHandle. 

Also, should name this method "readCustomInitializeApiDefinition" - "Job" is not an appropriate name here.